### PR TITLE
infra: switch to @stylistic/eslint-plugin

### LINKED ISSUE #3068: Switching to stylistic eslint plugin
As `typescript-eslint` v8 removes some rules we are using in favor of https://eslint.style/, we want to implement e.g. `@stylistic/eslint-plugin-ts` so we can still benefit from rules like [`padding-line-between-statements`](https://eslint.style/rules/default/padding-line-between-statements)

This needs to be done, before we can merge #3064, so we don't loose the checks for these rules.

This issue was created out of a thread over in #3064:

    **todo (blocking):** before simply removing this lint rule, we need to add https://eslint.style/rules/ts/padding-line-between-statements

_Originally posted by @Shinigami92 in https://github.com/faker-js/faker/pull/3064#discussion_r1722393401_

In typescript-eslint v8 has been removed as it has been moved to the eslint-stylistic package.
This issues serves as a reminder to add the rule back and to unblock the dependency update: https://github.com/faker-js/faker/pull/3064

- https://github.com/faker-js/faker/pull/3064

### PR DESCRIPTION
**Pull request by Shinigami92**
closes #3068 

unblocks #3064

### CODE REVIEW

File: eslint.config.js [hunk 1]
```diff
@@ -129,6 +126,20 @@ export default tseslint.config(
   },
   //#endregion
 
+  //#region stylistic
+  {
+    plugins: {
+      '@stylistic': stylistic,
+    },
+    rules: {
+      '@stylistic/semi': 'error',
```
Comment (Shinigami92) [inline eslint.config.js]: **thought:** not yet sure, if we should re-enable `semi`, as this is already covered by prettier.  
We could decide to only enable rules which are not covered by prettier.  
Maybe discuss this in tomorrows meeting?

**Review by Shinigami92 — COMMENTED** [review]
tested locally by removing some blank lines, and seems to work fine  
I followed the migration guide on their docs and decided to take the "easy" path and use the full combined plugin variant, where we can enable more rules in the future  
It looks like there are no "recommended" rule sets as in other plugins, because their directive is more like "these are stylistic rules which are turned on individually"

File: eslint.config.js (hunk 1 above)
Comment (Shinigami92) [reply eslint.config.js]: we dont need to configure already prettier formatted things

```suggestion
```

File: eslint.config.js [hunk 2]
```diff
@@ -129,6 +126,19 @@ export default tseslint.config(
...
+    plugins: {
+      '@stylistic': stylistic,
+    },
+    rules: {
+      '@stylistic/padding-line-between-statements': [
+        'error',
+        { blankLine: 'always', prev: 'block-like', next: '*' },
+      ],
+    },
+  },
```
Comment (ST-DDT) [inline eslint.config.js:140]: This is missing the endregion comment.

File: eslint.config.js [hunk 3]
```diff
@@ -129,6 +126,19 @@ export default tseslint.config(
...
+
```
Comment (Shinigami92) [inline eslint.config.js:142]: **suggestion:** 
```suggestion
  //#endregion

```

File: eslint.config.js (hunk 2 above)
Comment (Shinigami92) [reply eslint.config.js:140]: nice catch 💪

**Review by xDivisionByZerox — APPROVED** [review]

**Review by ST-DDT — APPROVED** [review]

**Pull request merged**